--- unzip60/unzpriv.h.orig	2010-02-18 01:03:46.747893369 +0800
+++ unzip60/unzpriv.h	2010-02-18 01:06:53.797865684 +0800
@@ -2900,7 +2900,7 @@
 #    endif
 #    define _ISO_INTERN(str1) if (iso2oem) {register uch *p;\
        for (p=(uch *)(str1); *p; p++)\
-         *p = native((*p & 0x80) ? iso2oem[*p & 0x7f] : *p);}
+         *p = native(*p);}
 #  else
 #    define _ISO_INTERN(str1)   A_TO_N(str1)
 #  endif
@@ -2915,7 +2915,7 @@
 #    endif
 #    define _OEM_INTERN(str1) if (oem2iso) {register uch *p;\
        for (p=(uch *)(str1); *p; p++)\
-         *p = native((*p & 0x80) ? oem2iso[*p & 0x7f] : *p);}
+         *p = native(*p);}
 #  endif
 #endif
 
